### ISSUE TITLE
Generative model trained on Vizwiz dataset

### ISSUE THREAD

**Issue by shubhamagarwal92** [facebookresearch/mmf#228]
## ❓ Questions and Help

Hi @apsdehal,

Could you please tell as to how we can directly use a generative model pre-trained on Vizwiz dataset in the evaluation mode. 

Basically, we want to have a service, similar to the Visual dialog [demo](http://demo-visualdialog.cloudcv.org/) where we can pass directly the image and the question (w/- historical context) through a python script. 

Thanks

**Comment by apsdehal** [comment]
Look into how captioning evaluation works. Something similar will need to be evaluated with greedy/beam search.

**Comment by shubhamagarwal92** [comment]
1. Could you point me to the code you are referring to? Can we use it out of the box with minimal changes for deployment? 
2. I am guessing that the pretrained Pythia model for Vizwiz [here](https://learnpythia.readthedocs.io/en/latest/tutorials/pretrained_models.html) is the discriminative one requiring BUTD features? Is this Pythia model the same as described in this technical [report](https://arxiv.org/pdf/1807.09956.pdf)? For using the generative model, we would have to train the decoder?
3. To use the model in 2. above in a real-life setting, we would have to pass it through the whole pipeline and first extract the object-level features and then call the evaluation script. Is there any running example of this pipeline somewhere?
4. Do you suggest to call this kind of [inference](https://learnpythia.readthedocs.io/en/latest/tutorials/quickstart.html#inference) for the VizWiz task?

**Comment by apsdehal** [comment]
1. See [this](https://github.com/facebookresearch/pythia/blob/master/pythia/tasks/captioning/coco/dataset.py#L56) and [this](https://github.com/facebookresearch/pythia/blob/master/pythia/models/butd.py#L153).
2. Yes, it is the discriminative one but you should be able to convert it to a generative one following the BUTD captioning example. Yes, it is the same Pythia model, but it is pretrained on VQA2 and fine-tuned on VizWiz.
3. See the demo pipeline in colab notebook mentioned in README.
4. No, see the demo pipeline. That's how you should do inference.

**Comment by shubhamagarwal92** [comment]
Thanks for your pointers :)
